gems/sorbet-runtime/lib/types/types/typed_hash.rb

```diff
@@ -3,22 +3,32 @@
 
 module T::Types
   class TypedHash < TypedEnumerable
-    # Technically we don't need these, but they are a nice api
-    attr_reader :keys, :values
-
     def underlying_class
       Hash
     end
 
     def initialize(keys:, values:)
-      @keys = T::Utils.coerce(keys)
-      @values = T::Utils.coerce(values)
-      @type = T::Utils.coerce([keys, values])
+      @inner_keys = keys
+      @inner_values = values
+    end
+
+    # Technically we don't need these, but they is a nice api
```

Comment:
```suggestion
    # Technically we don't need these, but they are a nice api
```

Reply:
I actually changed the comment for using the singular form since I copied it other two separate methods 👍